docs/src/userguide/language_basics.rst

```diff
@@ -208,6 +208,50 @@ In the interests of clarity, it is probably a good idea to always be explicit
 about object parameters in C functions.
 
 
+Optional Arguments
+------------------
+
+Unlike C, it is possible to use optional arguments in ``cdef`` and ``cpdef`` functions.
+There are differences though whether you declare them in a ``.pyx`` file or a ``.pxd`` file.
```

Comment:
"or a .pxd file" -> "or the corresponding .pxd file"
Add: "To avoid repetition (and potential future inconsistencies), default argument values are not visible in the declaration (in .pxd files) but only in the implementation (in .pyx files)."